server/coordinator/queue/delay_queue.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package queue
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/CeresDB/ceresmeta/server/coordinator/procedure"
+	"github.com/pkg/errors"
+)
+
+type ProcedureWithTime struct {
+	procedure procedure.Procedure
+	time      time.Time
+}
+
+type ProcedureDelayQueue struct {
+	lock       sync.RWMutex
+	procedures []*ProcedureWithTime
+	maxLen     int
+}
+
+func NewProcedureDelayQueue(maxLen int) *ProcedureDelayQueue {
+	return &ProcedureDelayQueue{
+		procedures: []*ProcedureWithTime{},
+		maxLen:     maxLen,
+	}
+}
+
+func (q *ProcedureDelayQueue) Len() int {
+	q.lock.RLock()
+	defer q.lock.RUnlock()
+
+	return len(q.procedures)
+}
+
+func (q *ProcedureDelayQueue) Offer(p procedure.Procedure, delaySec time.Duration) error {
```

Comment:
It is rare to name the methods of a Queue as Offer and Poll. How about Push and Pop?
```suggestion
func (q *ProcedureDelayQueue) Push(p procedure.Procedure, delay time.Duration) error {
```